custom-routes-vis/src/main/webapp/script.js

```diff
@@ -143,9 +156,140 @@ function hideMarker(markerName) {
   document.getElementById('generate-routes').style.display = 'none';
 }
 
+/** Removes all routes from the DOM. */
+function clearRoutes() {
+  for (let i = 0; i < displayedRoutes.length; i++) {
+    displayedRoutes[i].setMap(null);
+  }
+  displayedRoutes = [];
+  selectedRouteNum = 0;
+}
+
 /**
- * Generates routes from origin to destination by calling Directions API.
+ * Generates routes from origin to destination by calling the selected API.
+ * Precondition: two markers exist in the originDestinationMarkers array.
  */
 function generateRoutes() {
-  // TODO(chenyuz): Write this code in separate branch.
-}
\ No newline at end of file
+  clearRoutes();
+
+  // Get coordinates of origin and destination.
+  let startLatLng = originDestinationMarkers[0].getPosition();
+  let endLatLng = originDestinationMarkers[1].getPosition();
+  let origin = startLatLng.lat() + ',' + startLatLng.lng();
+  let destination = endLatLng.lat() + ',' + endLatLng.lng();
+
+  let serviceEndpoint = document.getElementById('service-endpoint').value;
+  let apiKey = document.getElementById('api-key').value;
+
+  // Get the routes and display them on map.
+  fetch('/get-directions?origin=' + origin + '&destination=' + destination
+      + '&endpoint=' + serviceEndpoint + '&apiKey=' + apiKey)
+      .then(response => response.json()).then(directions => {
+        // Log the response status from the Directions API.
+        // TODO(chenyuz): find equivalence for Routes Preferred API.
+        console.log(directions.status);
+
+        let routes = directions.routes;
+        console.log('num routes:', routes.length);
```

Comment:
Are these logging statements for debugging? Can we remove them?

Reply:
I think it's good to see the number of routes anyway